File: trunk/lucette2/src/main/java/battle/csn/lucette/machine/Chain.java
package battle.csn.lucette.machine;

import battle.csn.lucette.machine.states.InitialState;
import battle.csn.lucette.machine.states.framework.IState;


public class Chain
{
    private IState current;
    private FightStateMachine stateMachine;

    public Chain( FightStateMachine stateMachine )
    {
        this.setStateMachine( stateMachine );
        current = new InitialState( );
    }

    public void setState( IState s )
    {
        current = s;
    }

    public void pull( )
    {
        current.pull( this );
    }

    public boolean done( )
    {
        return current.done( );

    }

    public FightStateMachine getStateMachine( )
    {
        return stateMachine;
    }

    public void setStateMachine( FightStateMachine stateMachine )
    {
        this.stateMachine = stateMachine;
    }
}

File: trunk/lucette2/src/main/scala/battle/csn/lucette2/engine/Chain.scala
package battle.csn.lucette2.engine

import battle.csn.lucette2.engine.states.framework.State
import battle.csn.lucette2.engine.states.InitialState

class Chain (var stateMachine:FightStateMachine){
    var current:State = new InitialState( )

    def pull( ){
        current.pull( this )
    }

    def done( )={
        current.done( )
    }
}